silq/analysis/analysis.py

```diff
@@ -715,8 +717,44 @@ def analyse_EPR(empty_traces: np.ndarray,
             'mean_high_blip_duration': results_read['mean_high_blip_duration']}
 
 
-def analyse_flips(up_proportions_arrs: List[np.ndarray],
-                  threshold_up_proportion: Union[Sequence, float]):
+def analyse_threshold_up_proportion(up_proportions_arrs: np.ndarray,
+                                    shots_per_frequency: int):
+    if shots_per_frequency < 25:
+        num_shots = 25   # to make sure algorithm works
+    else:
+        num_shots = shots_per_frequency
+    proportion_space = np.linspace(0, 1, num=num_shots + 1, endpoint=True)
+    up_proportions_arrs = up_proportions_arrs.reshape(1, -1)
+    kernel = gaussian_kde(up_proportions_arrs)
+    gaussian_up_proportions = kernel(proportion_space)
+    peak_idxs = peakutils.peak.indexes(gaussian_up_proportions,
+                                       thres=0.5/up_proportions_arrs.shape[-1],
+                                       min_dist=num_shots/5)
+    if len(peak_idxs) == 0:
+        logger.debug(f'Adaptive thresholding routine: 0 peaks were '
+                     f'found, using threshold 0.5')
+        trough_idx = (num_shots + 1) // 2
+    elif len(peak_idxs) == 1:
```

Comment:
I can't easily understand what's happening in this section of code. It needs some comments, and preferably also one or two intermediate variables. For isntance, is there a variable name you could give to: `np.round(gaussian_up_proportions[trough_slice], 4)`?